src/main/java/net/minestom/server/tag/TagReader.java

```diff
@@ -0,0 +1,48 @@
+package net.minestom.server.tag;
+
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+import org.jglrxavpok.hephaistos.nbt.NBTCompound;
+
+/**
+ * Represents an element which can read {@link Tag tags}.
+ */
+public interface TagReader {
```

Comment:
I wonder if this should be named `TagReadable`. `TagReader` to me implied something which reads tags, not something from which tags can be read.